### DISCUSSION TITLE
Cannot remove DeDRM

### DISCUSSION THREAD

**Discussion by alrightson** [apprenticeharper/DeDRM_tools#1854] [Q&A]
Hi all,

I have tried the latest calibre and DeDRM aswell as Calibre 4.23 and Apprentice DeDRM 6.81 and converting books just fails.  I have even installed kindle for mac 1.73 on my sons MacBook Pro and had him send the .azw file over to me and it still fails.

Can anyone help please?

**Comment by ElleKayEm** [comment]
Each installation of the Kindle app has its own unique key.  So the app used to download the book needs to be on the same machine and user as calibre with DeDRM.  Also take a look at this post for dealing with (or avoiding) KFX format:  https://www.mobileread.com/forums/showthread.php?t=283371

And this has Mac specific info:  https://www.mobileread.com/forums/showpost.php?p=3819708&postcount=508

**Reply by alrightson** [reply]
Yeah followed that, and tried it again and when i tried to convert the book I get error saying it is locked and can't convert

**Comment by ElleKayEm** [comment]
Let's see a log:  https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

Use calibre 4.23 and DeDRM 6.8.1 if this is on a Mac.  They can have trouble initially retrieving the Kindle app's key with DeDRM 7.2.1; but once things are working, you should be able to update calibre and DeDRM and still decrypt books from that app.

**Reply by alrightson** [reply]
thank you for your help. After seeing the log it was still showing kfx-zip so had to do the chmod again. which has now worked. Thank you again

**Comment by ElleKayEm** [comment]
Good to hear!  Thanks for reporting back.
